dodder-common/src/test/java/cc/dodder/common/entity/TreeTest.java

```diff
@@ -0,0 +1,134 @@
+package cc.dodder.common.entity;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+
+public class TreeTest {
+
+    private final ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+    private final PrintStream output = System.out;
+
+    @Before
+    public void setUpStreams() {
+        System.setOut(new PrintStream(outputStream));
+    }
+
+    @After
+    public void restoreStreams() {
+        System.setOut(output);
+    }
+
+    @Test
+    public void testCheckExist() {
+        final Tree tree = new Tree();
+        final Node node = new Node();
+        node.addChild(new Node());
+
+        assertFalse(tree.checkExist(node, null));
```

Comment:
Is there not a test case for this that returns `true`, and if not, could you write one?

Reply:
No test case that returns `true` and as far as I can see there isnt a way to write one.

Reply:
I think you could write one :) you'd just have to work out how to construct each node so that it would return true

Reply:
Well i think the method is incorrect and is not actually able to return true as it seems to just search through the tree but never actually compares the string that is entered for the search. 
`	public boolean checkExist(Node tnode, String text) {`
`		boolean exist = false;`
`		if (tnode.getChildren() == null) {`
`			return false;`
`		}`
`		for (Node node : tnode.getChildren()) {`
`			exist |= checkExist(node, text);`
`		}`
`		return exist;`
`	}`

Reply:
In that case, that's something to point out in the PR then :)